File: src/main/java/io/playground/orderservice/application/order/port/OrderItemPersistencePort.java
package io.playground.orderservice.application.order.port;

import io.playground.orderservice.domain.order.Order;
import io.playground.orderservice.domain.order.OrderItem;

import java.util.List;
import java.util.Map;

public interface OrderItemPersistencePort {
    List<OrderItem> findAllByOrderExternalIdAndOrderStatus(String orderExternalId,
                                                           Order.OrderStatus orderStatus);
    List<OrderItem> findAllByVariantIdsAndOrderExternalIdAndOrderStatusIn(String orderExternalId,
                                                                          List<Long> variantIds,
                                                                          List<Order.OrderStatus> orderStatuses);

    void saveAll(List<OrderItem> orderItems);

    boolean updateCanceledReasonsToCancelAll(String orderExternalId,
                                             String reason);

    boolean updateCanceledInfosToCancelPartially(String orderExternalId,
                                                 Map<Long, Integer> cancelsItemQuantities,
                                                 String reason);
}

File: src/main/kotlin/io/playground/orderservice/application/order/port/OrderItemPersistencePort.kt
package io.playground.orderservice.application.order.port

import io.playground.orderservice.domain.order.Order
import io.playground.orderservice.domain.order.OrderItem

interface OrderItemPersistencePort {
    fun findAllByOrderExternalIdAndOrderStatus(
        orderExternalId: String,
        orderStatus: Order.OrderStatus,
    ): List<OrderItem>

    fun findAllByVariantIdsAndOrderExternalIdAndOrderStatusIn(
        orderExternalId: String,
        variantIds: List<Long>,
        orderStatuses: List<Order.OrderStatus>,
    ): List<OrderItem>

    fun saveAll(orderItems: List<OrderItem>)

    fun updateCanceledReasonsToCancelAll(
        orderExternalId: String,
        reason: String,
    ): Boolean

    fun updateCanceledInfosToCancelPartially(
        orderExternalId: String,
        cancelsItemQuantities: Map<Long, Int>,
        reason: String,
    ): Boolean
}
